app/services/applications/change_funding_eligibility.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Applications
+  class ChangeFundingEligibility
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    OPTIONS = {
+      true => I18n.t("shared.yes"),
+      false => I18n.t("shared.no"),
+    }.freeze
+
+    attribute :application
+    attribute :eligible_for_funding, :boolean
+
+    validates :application, presence: true
+    validates :eligible_for_funding, inclusion: OPTIONS.keys
+
+    validate  :validate_funding_eligiblity_status_code_change, if: :application
+    validate  :validate_funding_eligiblity_status_with_funded_place, if: :application
+
+    def eligible_for_funding_options
+      OPTIONS
+    end
+
+    def change_funding_eligibility
+      return false if invalid?
+
+      funding_eligiblity_status_code =
+        (eligible_for_funding ? :marked_funded_by_policy : :marked_ineligible_by_policy)
+
+      application.update!(eligible_for_funding:, funding_eligiblity_status_code:)
+    end
+
+  private
+
+    def validate_funding_eligiblity_status_code_change
+      if declared_as_billable? && eligible_for_funding == false
```

Comment:
I think if we have submitted declarations and we mark the eligible for funding as true, that means those declarations need to become eligible. I think worth checking with Jake the intended logic, should we allow this at all on applications with declarations, or should we change the declarations state themselves?